src/main/java/com/squareup/javapoet/Util.java

```diff
@@ -82,4 +154,23 @@ public static String join(String separator, List<String> parts) {
     result.addAll(b);
     return result;
   }
+
+  public static Set<Modifier> getMissingImplicitModifiers(Set<Modifier> modifiers,
```

Comment:
Hmmmm I don't know if I follow this. It's kind of hard to reason about what's happening here, with a whole bunch of different sets. Wouldn't we be better off at the call sites to have a method like this:

```
  if (kind == INTERFACE) {
    requireSubsetOf(modifiers, Modifier.PUBLIC, Modifier.ABSTRACT, Util.DEFAULT);
    requireExactlyOne(modifiers, Modifier.ABSTRACT, Util.DEFAULT);
  }
```

As is, it's actually awkward to recognize what the rules are.